arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -246,26 +246,48 @@ ::filter( kwiver::vital::image_container_sptr input_image )
 
   std::vector< vil_image_view< pix_t > > filtered_images;
 
-  if( enable_color || enable_gray )
-  {
-    const auto vxl_image =
-      convert_to_typed_vil_image_view< pix_t >( input_image );
+  vil_image_view< double > double_variance;
 
+  if( enable_color )
+  {
     // 3 channels
-    if( enable_color )
+    auto color = convert_to_typed_vil_image_view< pix_t >(
+      input_image );
+    filtered_images.push_back( color );
+  }
+
+  // These three features require processing the vil_image directly
+  if( enable_gray || enable_average || enable_normalized_variance )
+  {
+    auto input_image_sptr =
+      vxl::image_container::vital_to_vxl( input_image->get_image() );
+
+    if( input_image_sptr->nplanes() == 3 )
+    {
+      input_image_sptr = vil_convert_to_grey_using_rgb_weighting(
+        input_image_sptr );
+    }
+    else
```

Comment:
I'm not sure this is necessary, as the VIL function may have the same optimization, but maybe for clarity? What do you think?
```suggestion
    else if( input_image_sptr->nplanes() > 1 )
```

Reply:
`vil` does the same thing [here](https://vxl.github.io/doc/release/core/vil/html/vil__convert_8h_source.html#l00974) as best I can tell. I don't personally feel this adds much value, but I could be convinced if you think so.

Reply:
Nah, that's what I was asking. I think it comes down more to if it's clearer to the reader that we don't do anything if the input is already single-channel. If you don't feel the suggested change improves code readability, feel free to ignore it :slightly_smiling_face:.